src/mod/databases/mod_mariadb/mod_mariadb.c

```diff
@@ -627,13 +627,22 @@ switch_status_t mariadb_send_query(mariadb_handle_t *handle, const char* sql)
 {
 	char *err_str;
 	int ret;
+	unsigned retries = 100;
 
 	switch_safe_free(handle->sql);
 	handle->sql = strdup(sql);
+    again:
 	handle->stored_results = 0;
 	ret = mysql_real_query(&handle->con, sql, (unsigned long)strlen(sql));	
 	if (ret) {
 		err_str = mariadb_handle_get_error(handle);
+		if (strstr(err_str, "Deadlock found when trying to get lock; try restarting transaction")) {
+			if (--retries > 0) {
+			    switch_safe_free(err_str);
+			    goto again;
```

Comment:
Changed the code.
Added a database wait for a time from 500 to 1000 ms.
Random locking is needed in case 2 instances of freeswitch work in parallel and simultaneously try to work with the same table